src/stores/store/secrets/hashicorp/hashicorp.go

```diff
@@ -202,16 +209,19 @@ func (s *Store) Restore(_ context.Context, id, version string) error {
 	return nil
 }
 
-func (s *Store) Destroy(_ context.Context, id, version string) error {
-	if version == "" {
-		err := errors.NotSupportedError("secret version must be specified to destroy")
-		s.logger.Warn(err.Error())
-		return err
-	}
-
+func (s *Store) Destroy(ctx context.Context, id string) error {
 	logger := s.logger.With("id", id)
-	err := s.client.WritePost(s.pathDestroyID(id), map[string][]string{
-		"versions": {version},
+	
+	versionList, err := s.db.ListVersions(ctx, id, false)
```

Comment:
this block can be moved to a private function given that it logs so you can return the err directly and avoid duplicating the errMessage definition. Linting should fail anyways.